react-events/my-app/src/App.tsx

```diff
@@ -0,0 +1,57 @@
+import { Header } from './Header';
+import { Image } from './Image';
+import { Caption } from './Caption';
+import { Description } from './Description';
+import { Button } from './Button';
+import { useState } from 'react';
+import './App.css';
+
+const srcs = ['/starry-sky.jpeg', '/cool-kitty.jpg', '/cool-car.jpg'];
+const captions = ['A Beautiful Image of Space', 'A Cool Cat', 'A Cool Car'];
+const descriptions = [
+  'This is a wonderful sky full of stars!',
+  'This is a picture of a very cool cat!',
+  'This is a picture of a very cool car!',
+];
+
+export function App() {
+  const [captionIndex, setCaptionIndex] = useState(0);
```

Comment:
This is a clever solution that shows you put some good thought into it. It's more complex than it needs to be, though. If you keep just one state variable (say, `index`), then you can use that same state variable for the image, caption, and description. So you really need just that one state variable and just one handler function.